### ISSUE TITLE
Request to add "date_received" field to TaskResult model 

### ISSUE THREAD

**Issue by amirzia** [celery/django-celery-results#148]
## Problem

Tasks can be scheduled and be [executed in the future](https://docs.celeryproject.org/en/master/userguide/calling.html#eta-and-countdown). When we call a scheduled task its state becomes `RECEIVED` after going through `PENDING` state. Until the time to execute arrives, the task would not be added into results database. Also, we don't have access to the time when the task was submitted (received by a worker).

## Proposed Solution

I suggest that a field named `date_received` (or something like that) be added to `TaskResult` model. Also, the scheduled task would be added to results db so that it can be tracked. Obviously, `date_received` is equal to `date_created` in case a task is not scheduled.

This feature is helpful because, in addition to knowing when a task started executing (`date_created `) and finished (`date_done`), we can inspect when it was submitted or received by a worker (`date_received`).

**Comment by auvipy** [comment]
> I suggest that a field named `date_received` (or something like that) be added to `TaskResult` model. Also, the scheduled task would be added to results db so that it can be tracked. Obviously, `date_received` is equal to `date_created` in case a task is not scheduled.
> 
> This feature is helpful because, in addition to knowing when a task started executing (`date_created `) and finished (`date_done`), we can inspect when it was submitted or received by a worker (`date_received`).

are you able to implement this? do we need any check for the field?

**Comment by amirzia** [comment]
Yes, I can implement that. What do you mean by "check for the field"?

**Comment by auvipy** [comment]
I suggest you come with a proof of concept PR
